From efc24229b0ba4b2f6a39fe89a4c9c576dbe7e124 Mon Sep 17 00:00:00 2001
From: "David Kaspar [Dee'Kej]" <dkaspar@redhat.com>
Date: Mon, 27 Nov 2017 10:30:52 +0100
Subject: [PATCH] Bug 698784: Fix the alias for Helvetica-Narrow-Bold-Oblique

Init/Fontmap.GS:

There was a misalignment between the filename of the
NimbusSansNarrow-BdOblique and its /Fontname in the T1 font itself.

Filename:   NimbusSansNarrow-BdOblique
/Fontname: /NimbusSansNarrow-BoldOblique

This worked correctly if the fonts were located directly in
/usr/share/ghostcript/Resources/Font folder on the filesystem.

However, on Fedora we are using /usr/share/fonts/urw-base35/ folder,
which is part of Ghostscript's Search Path. In this case it was
causing the Ghostscript being unable to locate the correct font, thus
crashing while opening any document containing Helvetica Narrow Bold
Oblique font...

*In addition*, rename the font file so it matches.
---
 ...busSansNarrow-BdOblique => NimbusSansNarrow-BoldOblique} | Bin
 Resource/Init/Fontmap.GS                                    |   2 +-
 psi/psromfs.mak                                             |   2 +-
 3 files changed, 2 insertions(+), 2 deletions(-)
 rename Resource/Font/{NimbusSansNarrow-BdOblique => NimbusSansNarrow-BoldOblique} (100%)

diff --git a/Resource/Font/NimbusSansNarrow-BdOblique b/Resource/Font/NimbusSansNarrow-BoldOblique
similarity index 100%
rename from Resource/Font/NimbusSansNarrow-BdOblique
rename to Resource/Font/NimbusSansNarrow-BoldOblique
diff --git a/Resource/Init/Fontmap.GS b/Resource/Init/Fontmap.GS
index 7770c67..cbda218 100644
--- a/Resource/Init/Fontmap.GS
+++ b/Resource/Init/Fontmap.GS
@@ -96,7 +96,7 @@
 /Helvetica-Bold			/NimbusSans-Bold ;
 /Helvetica-BoldOblique		/NimbusSans-BoldItalic      ;
 /Helvetica-Narrow-Bold		/NimbusSansNarrow-Bold    ;
-/Helvetica-Narrow-BoldOblique	/NimbusSansNarrow-BdOblique        ;
+/Helvetica-Narrow-BoldOblique	/NimbusSansNarrow-BoldOblique        ;
 /Helvetica-Narrow		/NimbusSansNarrow-Regular    ;
 /Helvetica-Narrow-Oblique	/NimbusSansNarrow-Oblique        ;
 /Helvetica			/NimbusSans-Regular ;
diff --git a/psi/psromfs.mak b/psi/psromfs.mak
index cbb1163..34e5b5f 100644
--- a/psi/psromfs.mak
+++ b/psi/psromfs.mak
@@ -333,7 +333,7 @@ PS_FONT_DEPS=\
 	$(PSRESDIR)$(D)Font$(D)NimbusRoman-Regular \
 	$(PSRESDIR)$(D)Font$(D)NimbusSans-Bold \
 	$(PSRESDIR)$(D)Font$(D)NimbusSans-BoldItalic \
-	$(PSRESDIR)$(D)Font$(D)NimbusSansNarrow-BdOblique \
+	$(PSRESDIR)$(D)Font$(D)NimbusSansNarrow-BoldOblique \
 	$(PSRESDIR)$(D)Font$(D)NimbusSansNarrow-Bold \
 	$(PSRESDIR)$(D)Font$(D)NimbusSansNarrow-Oblique \
 	$(PSRESDIR)$(D)Font$(D)NimbusSansNarrow-Regular \
-- 
2.9.5

